### PR TITLE
win midi: Fix crash in the sound menu of the setup without config

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

**Review by fabiangreffrath — APPROVED** [review]

File: src/i_winmusic.c [hunk 1]
```diff
@@ -49,7 +49,7 @@ enum
     RESET_TYPE_XG,
 };
 
-char *winmm_midi_device = NULL;
+char *winmm_midi_device = "";
```
Comment (turol) [inline src/i_winmusic.c:52]: This causes a constness issue.

File: src/i_winmusic.c [hunk 2]
```diff
@@ -1567,7 +1567,10 @@ static boolean I_WIN_InitMusic(void)
 
     if (i == all_devices)
     {
-        free(winmm_midi_device);
+        if (strlen(winmm_midi_device) > 0)
```
Comment (turol) [inline src/i_winmusic.c:1570]: This could theoretically cause a memory leak if the user-specified midi device is an empty string.

**Review by turol — CHANGES_REQUESTED** [review]
I think a better way would be to leave the initialization as is and explicitly check the variable against `NULL` where required.

File: src/i_winmusic.c (hunk 2 above)
Comment (rfomin) [reply src/i_winmusic.c]: Fixed in https://github.com/chocolate-doom/chocolate-doom/pull/1642/commits/7929bbc56c036a0261b5d28bdf4782d0e7bb6098

File: src/i_winmusic.c (hunk 1 above)
Comment (rfomin) [reply src/i_winmusic.c]: Fixed in https://github.com/chocolate-doom/chocolate-doom/pull/1642/commits/7929bbc56c036a0261b5d28bdf4782d0e7bb6098

**Review by turol — APPROVED** [review]
Should be squashed.

**Pull request merged**